src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -815,6 +816,47 @@ public Task RemoveProfileAsync(string profileName)
             });
         }
 
+        public Task UpdateProfileAsync(string profileName, Action<IWritableLaunchProfile> updateAction)
+        {
+            // Updates need to be sequenced
+            return _sequentialTaskQueue.ExecuteTask(async () =>
+            {
+                ILaunchSettings currentSettings = await GetSnapshotThrowIfErrors();
```

Comment:
Can we change `GetSnapshotThrowIfErrors` to have an `Async` suffix?

Reply:
Yes.